src/extensions/interpolation/syntax.ts

```diff
@@ -78,19 +91,19 @@ function tokenizeInterpolation(effects: Effects, ok: State, nok: State) {
       return onInterpolationFormula;
     }
 
-    return nok;
+    return nok(code);
   }
 
   function onInterpolationEnd(code: Code) {
+    // if the final code is '}', interpolation is complete,
+    // if it is anything else, we abort with nok
     if (code === codes.rightCurlyBrace) {
       effects.consume(code);
-      markers -= 1;
-
-      if (markers) return onInterpolationEnd;
```

Comment:
why are these removed?

this was previously used to ensure that the interpolation ended with the exact number of `}` (2 characters)

Reply:
I removed it because this is now done implicitly.

```js
  function onInterpolationFormula(code) {
    if (code === codes.rightCurlyBrace) {
      effects.exit(type);
      effects.enter("void");
      effects.consume(code);

      return onInterpolationEnd;
    }
```

the first `}` is already consumed by the time we call `onInterpolationEnd`. Therefore if we encounter this character within `onInterpolationEnd`, it is already the second instance, and if we encounter anything else (like in `{{f}x`) we should terminate. Beforehand, we would return `onInterpolationEnd` from the aforementioned if block without consuming, but this is also not allowed by micromark.